refactor: add explicit return types to functions

bajpai244 opened the pull request.

Kolezhniuk reviewed (changes requested):
Please make sure that unit tests pass and fix linter warnings

bajpai244 commented:
@Kolezhniuk , the tests didn't pass because they took more time than the timeout for this flow, I have fixed the timeout so it should work now.

- I have refactored to fix linter issues:
- There were console uses in some places {no-console} and they have been ignored via .eslintrc, I will later make a PR to remove the console and write the result of the function to some other customizable logger instead.
- There was also a minor bug in the implementation of nodeValue, which I have fixed.

Kolezhniuk commented on .eslintrc:
```diff
@@ -13,7 +13,7 @@
   "rules": {
     "@typescript-eslint/no-var-requires": 0,
-    "no-console": 1,
+    "no-console": 0,
```
Please revert this and put the ignoring rule in the file where console.log using

bajpai244 replied:
Done, [link to commit](https://github.com/iden3/js-merkletree/commit/c9dfa45fde67810ec81c97c42db7e4b15cf3ebb7)

bajpai244 commented on src/lib/db/inMemory.ts:
```diff
@@ -19,22 +19,22 @@ export default class inMemmoryDB implements Storage {
   }
 
-  async put(k: Bytes, n: Node) {
+  async put(k: Bytes, n: Node): Promise<void> {
```
@Kolezhniuk, the reason why it contains async/await right now is that when there would be a DB implementation in some production database like SQL, then, these methods will be asynchronous. 

The database is passed an input to the SMT constructor, and hence all DB implementations should implement the same interface, which in our case is [Storage](https://github.com/iden3/js-merkletree/blob/13584c6d8c3157354156555e8d4afdfda727a924/src/types/storage.ts#L5).

Let me know your thoughts on this 🙌

bajpai244 commented on .github/workflows/ci.yaml:
```diff
@@ -5,7 +5,7 @@ jobs:
       matrix:
         version: [16.15, 18]
-    timeout-minutes: 7
+    timeout-minutes: 30
```
I also pondered the same, on my m1 they are done within 10 minutes { sometime 5-7 }, The test suite is very long! and the Poseidon hashing algorithm we are using is in pure js implementation.

I think this could be the reason.
